tests/test-kicker.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# This test exercises the kicker script functionality by reading in sample data
+# containing 15 (defanged) ch202 entries and writing them each to a separate file
+# in a temporary directory.  For each file, we echo its name into test-kicker-json.out
+# (followed by a newline), and remove it.  Once all files have been kicked
+# (resulting in 15 lines in test-kicker-json.out) the total line count must equal 15.
+
+script_dir=$( cd -- "$( dirname -- "${BASH_SOURCE[0]}" )" &> /dev/null && pwd )
+nmsgtool_test=$script_dir/../src/nmsgtool
+infile=$script_dir/private-tests/202_sample.json
+outdir=/tmp/test-kicker/ # This directory will get rm -rf'd; be careful!
+outfile=$outdir/test-kicker-json.out
+kicker="echo"
+badkicker="echo && rm"
+retval=0
+
+check() {
+	if [ $? = "0" ]; then
+		echo "PASS: $*"
+	else
+		echo "FAIL: $*"
+		retval=1
+	fi
+}
+
+echo "Testing kicker: "
+
+mkdir $outdir
+cd $outdir
+# If we don't set this, make check will be unable to find this directory!
+NMSG_MSGMOD_DIR=$script_dir/../nmsg/base/.libs
+
+# Read input data with our kicked nmsgtools.
+$nmsgtool_test -ddddd -j $infile -c 1 -k "$kicker" > $outfile
+
+# Compare echo'd file count and actual file count with the desired file count.
+file_count=$(wc -l $outfile | awk '{print $1}')
+true_file_count=$(ls -1 $outdir | wc -l)
+[[ "$file_count" == "15" ]] && [[ "$true_file_count" == "16" ]] # 16 to account for json output file.
+check "comparison of kicked json output files"
+
+# Remove kicked files from $outdir.
+while IFS= read -r line; do
+	rm $line
+done < $outfile
+
+# Ensure that a bad kicker script does nothing!  Thank you, quotation marks.
+$nmsgtool_test -ddddd -j $infile -c 1 -k $badkicker > $outfile
```

Comment:
Note that this attempt without quotes produces the following logs:

```
nmsgtool: usage problem, 2 unwanted additional arguments
Usage: nmsgtool
[-b filter]       or --bpf        filter pcap inputs with this bpf
[-B byterate]     or --byterate   ingress byte rate limit for file input
[-c count]        or --count      stop or reopen after count payloads output
```

So in reality nothing is being tested. The test is not even getting past the getopt gateway.

Reply:
Feeling a little silly about this one.  Fixed!